tests/support/mod.rs

```diff
@@ -6,197 +6,142 @@ Test supports module.
+            NonZeroU32::new(800).unwrap(),
+            NonZeroU32::new(600).unwrap(),
+        );
+
+        let surface = unsafe { gl_config.display().create_window_surface(&gl_config, &attrs).unwrap() };
+
+        request.send((window, not_current_gl_context, surface)).unwrap();
     }
 }
 
-// SAFETY
-// requires `From` implementation to be kept in sync with `raw_window_handle` and `winit` crates
-unsafe impl Send for HandleOrWindow {}
-
-unsafe fn initialize_event_loop() {
-    INIT_EVENT_LOOP.call_once(|| {
-        // One-time-use channel to get the event loop proxy
-        let (ots, otr) = std::sync::mpsc::sync_channel(0);
-        // Transfers window and config for creating display
-        let (sender, receiver) = std::sync::mpsc::channel();
-
-        let builder = thread::Builder::new().name("event_loop".into());
-        builder
-            .spawn(|| {
-                let event_loop_res = if cfg!(unix) || cfg!(windows) {
-                    EventLoop::builder().with_any_thread(true).build()
-                } else {
-                    EventLoop::builder().build()
-                };
-                let event_loop = event_loop_res.expect("event loop building");
-                let proxy = event_loop.create_proxy();
-
-                #[allow(deprecated)]
-                event_loop.run(move |event, window_target| {
-                    match event {
-                        Event::UserEvent(_) => {
-                            let window_attributes = Window::default_attributes().with_visible(false);
-                            let config_template_builder = ConfigTemplateBuilder::new();
-                            let display_builder =
-                                DisplayBuilder::new().with_window_attributes(Some(window_attributes));
-                            let (window, gl_config) = display_builder
-                                .build(window_target, config_template_builder, |mut configs| {
-                                    // Just use the first configuration since we don't have any special preferences here
-                                    configs.next().unwrap()
-                                })
-                                .unwrap();
-
-                            // Leak the window object to obtain a static reference
-                            let boxed_window = Box::new(window.unwrap());
-                            let window = Box::leak(boxed_window);
-
-                            sender.send(((&*window).into(), gl_config)).unwrap();
-                        }
-                        _ => {
-                            // Send event loop proxy ASAP
-                            SEND_PROXY.call_once(|| {
-                                ots.send(proxy.clone()).unwrap();
-                            });
-                        }
-                    }
-                })
-                .unwrap();
-            })
-            .unwrap();
+pub struct WindowDisplay {
```

Comment:
The window can be kept alive in the app, but sending it to the test thread improves the DX on Wayland as non-visible windows are still [unsupported](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.set_visible).

Using a smart pointer instead of a record or tuple avoids updating the `build_display` call sites and is otherwise unnecessary.